server/routes/api/documents/schema.ts

```diff
@@ -273,17 +278,24 @@ export type DocumentsUpdateReq = z.infer<typeof DocumentsUpdateSchema>;
 export const DocumentsMoveSchema = BaseSchema.extend({
   body: BaseIdSchema.extend({
     /** Id of collection to which the doc is supposed to be moved */
-    collectionId: z.string().uuid(),
+    collectionId: z.string().uuid().optional(),
 
     /** Parent Id, in case if the doc is moved to a new parent */
     parentDocumentId: z.string().uuid().nullish(),
 
     /** Helps evaluate the new index in collection structure upon move */
     index: z.number().gte(0).optional(),
+
+    /** Whether this should be considered a template */
+    template: z.boolean().optional(),
```

Comment:
I don't understand this option or how it relates to moving

Reply:
This provides an additional validation to ensure only templates can be moved into a collection-less space.

Umm.. It's not used in the route impl; perhaps this is an overkill..?

Reply:
I think this is not needed. The correct way to perform this validation is after fetching the doc from the database.

Reply:
> The correct way to perform this validation is after fetching the doc from the database

👍